R/matchDim.R

```diff
@@ -0,0 +1,53 @@
+#' match object dimensions of a magclass object to the dimensions of a reference object
+#'
+#' A helper that restricts and expands a magclass object x to the size of a magclass object ref.
+#' Dimension names not present in x are added and set to the value provided by 'fill'.
+#' Dimension names not present in ref are cropped.
+#'
+#' @param x a magclass object to be modified
+#' @param ref a magclass object used as a reference for the modification
+#' @param fill value to be set in new dimensions
+#' @param dim subset of dimensions for which the matching should be done.
+#' Can be either a number between 1 and 3 or a vector of these.
+#' Set to NULL (default) for matching all dimensions.
+#' @author Falk Benke
+#'
+#' @export
+matchDim <- function(x, ref, dim = NULL, fill = NA) {
+
+  if (is.null(dim)) {
+    dim <- c(1, 2, 3)
+  } else if (length(setdiff(dim, c(1, 2, 3))) > 0) {
+    stop("argument 'dim' can only contain numbers between 1 and 3")
+  }
+
+  if (3 %in% dim && ndim(x, dim = 3) != ndim(ref, dim = 3)) {
```

Comment:
you need the same check for all the other dimensions as all main dimensions nowadays can have subdimensions (see for instance `maxample("animal")`)